plugin/output/file/file_test.go

```diff
@@ -0,0 +1,382 @@
+	cfg.FileName = file[0 : len(file)-len(cfg.FileExtension)]
+
+	p := Plugin{config: &cfg}
+
+	for _, tc := range testsCases {
+
+		//create files
+		files := make([]*os.File, len(tc.filesName))
+		createDir(t, p.config.TargetDir)
+		defer clearDir(t, p.config.TargetDir)
+
+		for _, f := range tc.filesName {
+			files = append(files, createFile(t, f, nil))
+		}
+		//	make check
+		idx := p.getStartIdx()
+		assert.EqualValues(t, tc.expectedIdx, idx)
+
+		//	close files
+		for _, f := range files {
+			f.Close()
+		}
+		clearDir(t, p.config.TargetDir)
+
+	}
+}
+
+func TestSealUpHasContent(t *testing.T) {
+	cfg := Config{
+		TargetFile:         targetFile,
+		RetentionInterval_: time.Second,
+		Layout:             "01",
+		FileMode_:          0666,
+
+		TargetDir:     "",
+		FileExtension: "",
+		FileName:      "",
+	}
+
+	dir, file := filepath.Split(cfg.TargetFile)
+
+	cfg.TargetDir = dir
+	cfg.FileExtension = filepath.Ext(file)
+	cfg.FileName = file[0 : len(file)-len(cfg.FileExtension)]
+
+	createDir(t, cfg.TargetDir)
+
+	defer clearDir(t, cfg.TargetDir)
+
+	d := []byte("some data")
+	f := createFile(t, cfg.TargetFile, &d)
+	defer f.Close()
+	p := Plugin{
+		config: &cfg,
+		mu:     &sync.RWMutex{},
+		file:   f,
+	}
+
+	infoInitial, _ := f.Stat()
+	assert.NotZero(t, infoInitial.Size())
+
+	//call func
+	p.sealUp()
+
+	//check work result
+	pattern := fmt.Sprintf("%s/%s*%s", cfg.TargetDir, cfg.FileName, cfg.FileExtension)
+	matches := getMatches(t, pattern)
+	assert.Equal(t, 2, len(matches))
+
+	//check new file was created and it is empty
+	info, err := p.file.Stat()
+	assert.EqualValues(t, 0, info.Size())
+	assert.NoError(t, err)
+
+	//check old file was renamed. And renamed file is not empty and contains data
+	for _, v := range matches {
+		if v != cfg.TargetFile {
+			info, err := os.Stat(v)
+			assert.NoError(t, err)
+			assert.EqualValues(t, len(d), info.Size())
+		}
+	}
+}
+
+func TestSealUpNoContent(t *testing.T) {
+	cfg := Config{
+		TargetFile:         targetFile,
+		RetentionInterval_: time.Second,
+		Layout:             "01",
+		FileMode_:          0666,
+
+		TargetDir:     "",
+		FileExtension: "",
+		FileName:      "",
+	}
+
+	dir, file := filepath.Split(cfg.TargetFile)
+
+	cfg.TargetDir = dir
+	cfg.FileExtension = filepath.Ext(file)
+	cfg.FileName = file[0 : len(file)-len(cfg.FileExtension)]
+
+	createDir(t, cfg.TargetDir)
+
+	defer clearDir(t, cfg.TargetDir)
+
+	f := createFile(t, cfg.TargetFile, nil)
+	defer f.Close()
+	p := Plugin{
+		config: &cfg,
+		mu:     &sync.RWMutex{},
+		file:   f,
+	}
+
+	infoInitial, _ := f.Stat()
+	assert.Zero(t, infoInitial.Size())
+
+	//call func
+	p.sealUp()
+
+	//check work result
+	pattern := fmt.Sprintf("%s/%s*%s", cfg.TargetDir, cfg.FileName, cfg.FileExtension)
+	assert.Equal(t, 1, len(getMatches(t, pattern)))
+
+	//check new file was created and it is empty
+	info, err := p.file.Stat()
+	assert.Zero(t, info.Size())
+	assert.NoError(t, err)
+}
+
+func TestStart(t *testing.T) {
```

Comment:
make this in e2e style, make retention short, write something, test that where is sealed file and other things